chpl/chpl-service/src/main/java/gov/healthit/chpl/permissions/domains/surveillance/UpdateActionPermissions.java

```diff
@@ -14,13 +29,25 @@ public boolean hasAccess() {
 
     @Override
     public boolean hasAccess(Object obj) {
-        if (!(obj instanceof Long)) {
+        if (!(obj instanceof Surveillance)) {
             return false;
         } else if (getResourcePermissions().isUserRoleAdmin() || getResourcePermissions().isUserRoleOnc()) {
             return true;
         } else if (getResourcePermissions().isUserRoleAcbAdmin()) {
-            Long acbId = (Long) obj;
-            return isAcbValidForCurrentUser(acbId);
+            Surveillance surv = (Surveillance) obj;
+            CertifiedProductDTO listing = null;
+            try {
+                if (surv.getCertifiedProduct() != null && surv.getCertifiedProduct().getId() != null) {
+                    listing = cpDao.getById(surv.getCertifiedProduct().getId());
+                }
+            } catch (EntityRetrievalException ex) {
+                LOGGER.error("Could not find listing with ID " + surv.getCertifiedProduct().getId());
+            }
+            if (listing != null && listing.getCertificationBodyId() != null
+                    && isAcbValidForCurrentUser(listing.getCertificationBodyId())) {
+                return true;
+            }
+            return false;
         } else {
```

Comment:
Understood not part of your change, but if I'm reading this correctly, technically, this else is not needed. Returning false alone should suffice.